### PR TITLE
Fix DynamicFabric locations

### PR DESCRIPTION
**Pull request by aledsage**
- When locations are defined at the YAML top-level, then ensure the
  DynamicFabric finds those.

### CODE REVIEW

**Comment by ahgittin** [comment]
looks good; will merge, but also have some related follow-on tidies

File: brooklyn-server/core/src/main/java/org/apache/brooklyn/entity/group/DynamicFabricImpl.java [hunk 1]
```diff
@@ -112,9 +115,10 @@ public void start(Collection<? extends Location> locsO) {
         if (locsO!=null) {
             addLocations(locsO);
         }
-        
+        Collection<Location> locs = Collections.unmodifiableCollection(Locations.getLocationsCheckingAncestors(getLocations(), this));
```
Comment (johnmccabe) [inline brooklyn-server/core/src/main/java/org/apache/brooklyn/entity/group/DynamicFabricImpl.java]: move this into the `newLocations.isEmpty()` condition we only need it it `locs0` is empty

File: brooklyn-server/core/src/main/java/org/apache/brooklyn/entity/group/DynamicFabricImpl.java [hunk 2]
```diff
@@ -18,11 +18,13 @@
  */
 package org.apache.brooklyn.entity.group;
 
+import static com.google.common.base.Preconditions.checkArgument;
```
Comment (johnmccabe) [inline brooklyn-server/core/src/main/java/org/apache/brooklyn/entity/group/DynamicFabricImpl.java]: we're already importing `com.google.common.base.Preconditions`

**Comment by ahgittin** [comment]
@johnmccabe i saw the same things, have applied them at #1161

**Pull request merged**